File: Modulo04/Heranca-CSharp/Professor.cs
using System;

public class Professor : Pessoa {
    private float salario;

    public Professor(string nome, int idade, float salario) : base(nome, idade) {
        this.salario = salario;
    }

    public float getSalario() {
        return this.salario;
    }

    public void setSalario(float salario) {
        this.salario = salario;
    }

    new public void imprime() {
        base.imprime();
        Console.WriteLine("Salario: {0}", this.salario);
    }
}

File: Modulo12/Professor-Cpp/Professor.cpp
#include "Professor.h"
#include <iostream>
using namespace std;

namespace poo {
    
    Professor::Professor(string nome, int CPF, double salario) : Pessoa(nome, CPF), salario(salario) {
    }

    Professor::~Professor() {
        cout << "Destrutor Professor: (" << salario << ")" << endl;
    }

    double Professor::getSalario() const {
        return salario;
    }

    void Professor::imprime() const {
        Pessoa::imprime();
        cout << "Salário: " << this->getSalario() << endl;
    }
    
}
